eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js

```diff
@@ -10,122 +11,194 @@ import Truncated from "components/Truncated";
 
-TD.propTypes = {
-  textAlign: PropTypes.oneOf(["left", "center", "right"]),
-};
-
-TD.defaultProps = {
-  textAlign: "left",
-};
+export const DuplicateQuestionnaireButton = styled(DuplicateButton)`
+  margin-right: 0.5em;
+`;
 
-class Row extends React.Component {
+export class Row extends React.Component {
   static propTypes = {
     questionnaire: CustomPropTypes.questionnaire.isRequired,
+    history: CustomPropTypes.history.isRequired,
     onDeleteQuestionnaire: PropTypes.func.isRequired,
     onDuplicateQuestionnaire: PropTypes.func.isRequired,
-    in: PropTypes.bool,
     exit: PropTypes.bool,
     enter: PropTypes.bool,
     autoFocus: PropTypes.bool,
   };
 
-  rowRef = React.createRef();
-
-  handleOpenDeleteQuestionnaireDialog = () => {
-    this.props.onDeleteQuestionnaire(this.props.questionnaire);
+  state = {
+    linkHasFocus: false,
+    disabled: true,
   };
 
-  handleDuplicateQuestionnaire = () => {
-    this.props.onDuplicateQuestionnaire(this.props.questionnaire);
-  };
-
-  shouldComponentUpdate(nextProps) {
-    for (let key of Object.keys(Row.propTypes)) {
-      if (this.props[key] !== nextProps[key]) {
-        return true;
-      }
-    }
-    return false;
-  }
+  rowRef = React.createRef();
 
   focusLink() {
     this.rowRef.current.getElementsByTagName("a")[0].focus();
   }
 
-  componentDidMount() {
-    if (this.props.autoFocus) {
+  componentDidUpdate(prevProps) {
+    if (!prevProps.autoFocus && this.props.autoFocus) {
       this.focusLink();
     }
   }
 
-  componentDidUpdate() {
-    if (this.props.autoFocus) {
-      this.focusLink();
-    }
-  }
+  handleClick = () => {
+    this.props.history.push(
+      buildQuestionnairePath({
+        questionnaireId: this.props.questionnaire.id,
+      })
+    );
+  };
+
+  handleFocus = () => {
+    this.setState({
+      linkHasFocus: true,
+    });
+  };
+
+  handleBlur = () => {
+    this.setState({
+      linkHasFocus: false,
+    });
+  };
+
+  handleButtonFocus = e => {
+    this.setState({
+      linkHasFocus: false,
+    });
+    e.stopPropagation();
+  };
+
+  handleLinkClick = e => {
+    e.stopPropagation();
+  };
+
+  handleDuplicateQuestionnaire = e => {
+    e.stopPropagation();
+    this.props.onDuplicateQuestionnaire(this.props.questionnaire);
+  };
+
+  handleDeleteQuestionnaire = e => {
+    e.stopPropagation();
+    this.props.onDeleteQuestionnaire(this.props.questionnaire);
+  };
 
   render() {
     const { questionnaire, ...rest } = this.props;
+
     return (
       <FadeTransition {...rest} exit>
-        <TR innerRef={this.rowRef}>
+        <TR
+          innerRef={this.rowRef}
+          onFocus={this.handleFocus}
+          onBlur={this.handleBlur}
+          linkHasFocus={this.state.linkHasFocus}
+          onClick={this.handleClick}
+        >
           <TD>
-            <TruncatedQuestionnaireLink
+            <QuestionnaireLink
               data-test="anchor-questionnaire-title"
               questionnaire={questionnaire}
```

Comment:
We don't need to pass this prop in now as `QuestionnaireLink` is just a styled component, and is being passed onto the DOM by that library. 

<img width="897" alt="Screenshot 2019-05-17 10 02 32" src="https://user-images.githubusercontent.com/930398/57918981-c27ae080-788f-11e9-8206-8368805fd5e3.png">